src/FunderMaps.Core/Services/ProductService.cs

```diff
@@ -125,11 +126,11 @@ public virtual async Task<IEnumerable<AnalysisProduct>> GetAnalysisByQueryAsync(
 
             // Process and return.
             // FUTURE Clean up, make async foreach or similar.
-            var result = new Collection<AnalysisProduct>();
-            foreach (var product in products)
+            var result = new ConcurrentBag<AnalysisProduct>();
+            Parallel.ForEach(products, (product) =>
             {
-                result.Add(await ProcessAnalysisAsync(userId, productType, product, token));
-            }
+                result.Add(Task.Run(() => ProcessAnalysisAsync(userId, productType, product, token)).Result);
+            });
```

Comment:
In case of `Task` there is another way to run parallel, see `Task.Factory.StartNew()`. No need to change anything, resolve this issue when read